torch/csrc/dynamo/guards.cpp

```diff
@@ -32,7 +32,8 @@ class TensorCheck {
         dispatch_key_(state.apply(v.key_set()).raw_repr()),
         dtype_(v.dtype().toScalarType()),
         device_index_(v.device().index()),
-        requires_grad_(state.grad_mode_enabled && v.requires_grad()),
+        requires_grad_(v.requires_grad()),
+        grad_mode_enabled_(state.grad_mode_enabled),
```

Comment:
It's weird that we are going to record this on every single TensorCheck lol

Reply:
yep that was Voz's feedback too - just updated, so we no longer stash grad mode state on the tensor guard, and instead we just have a single guard on the initial grad mode state when we start compiling.